### ISSUE TITLE
Loggin issue on DSM 7

### ISSUE THREAD

**Issue by NutTom** [seansfkelley/nas-download-manager#223]
### Description

Sometimes login doesn't work, had to log on DSM app with browser first to get the extension able to login then. Error is "unknown" without more information.

Be sending the request by hand (SYNO.API.Auth) it return error 103 ("The requested method does not exist" according to synology API documentation)

also according to synology's API (and many search on internet) : _In order to make API requests, you should first request SYNO.API.Info to get the SYNO.API.Auth API
info for session login and SYNO.DownloadStation.Task API info for download task list._

If I send first the SYNO.API.Info then I can then login with the extension and everything work great.
I checked the code and this function request is implemented but looks like it's never called before login.

### Steps to Reproduce the Issue

Try to login -> it fail
send the SYNO.API.Info request (or web login with brower)
login retry -> success

### Failure Logs

```
Check the bottom of the extension's setting page for a "Debugging Output" section and paste it over this text. If there is no such section in the settings, please say so.
```

### Versions

**Extension version**: 0.12.0

**Browser version**: chrome  100.0.4896.60

**DSM version**: DSM 7.0.1-42218 Update 3

**Comment by FCRX7** [comment]
Yes, same situation for me. Only recently started using NAS Download Manager (for Synology).
**Browser version:** Firefox 99.0 (64-bit Windows). 
**DSM version:** DSM 7.0.1-42218 Update 3 (three different Synology machines)
**Extension version:** 0.12.0

**Comment by NutTom** [comment]
Apparently it's the "connection test" function who need to be fixed by first making an info request to Synology, I'm trying to install everything to compile the code with my modification to help fixing this issue
